### PR TITLE
Test blocks

### LINKED ISSUE #1091: Network upgrade test blocks
It will be useful to have encoded block vectors for all the network upgrades so we can unit test the consensus changes for each. Related to https://github.com/ZcashFoundation/zebra/pull/1051

Ideally we should add 3 blocks per upgrade: NU-1, NU, and NU+1 for every network upgrade for mainnet and testnet.

This are the commands needed to be executed in a synchronized zcashd for blossom testnet(activation height: 584_000):

```
./src/zcash-cli getblock 583999 0
./src/zcash-cli getblock 584000 0
./src/zcash-cli getblock 584001 0
```

and blossom mainnet(activation height: 653_600)

```
./src/zcash-cli getblock 653599 0
./src/zcash-cli getblock 653600 0
./src/zcash-cli getblock 653601 0
```

Use https://github.com/ZcashFoundation/zebra/blob/main/zebra-chain/src/parameters/network_upgrade.rs#L44=L68 to get other heights.

TODO:

Networks
- [x] Mainnet
- [x] Testnet

Network Upgrades
- [x] Genesis
- [x] BeforeOverwinter
- [x] Overwinter
- [x] Sapling
- [x] Blossom
- [x] Heartwood
- [x] Canopy (Testnet only, until November)
  - [x] Open a follow-up ticket for Canopy Mainnet transition + 1 more Canopy Mainnet block (below the reorg limit) (#1099)

### PR DESCRIPTION
**Pull request by teor2345**
Closes #1091.

TODO:
- [x] Testnet Overwinter to Canopy

### CODE REVIEW

**Comment by teor2345** [comment]
@oxarbitrage I just did the mainnet blocks today - it took longer than I thought.

Hopefully that's enough to get you started.

**Comment by dconnolly** [comment]
@teor2345 is this still a draft?

File: zebra-test/src/vectors/block.rs [hunk 1]
```diff
@@ -0,0 +1,183 @@
+//! Block test vectors
+
+use hex::FromHex;
+use lazy_static::lazy_static;
+
+lazy_static! {
+    // Update this list of test blocks when you add a new block test vector to
+    // this file
+    pub static ref TEST_BLOCKS: Vec<&'static [u8]> = vec![
```
Comment (dconnolly) [inline zebra-test/src/vectors/block.rs:9]: 🧱🧱🧱

**Review by dconnolly — DISMISSED** [review]

**Comment by teor2345** [comment]
> @teor2345 is this still a draft?

Yeah it's missing the testnet blocks for each network upgrade - particularly Canopy, which we can't test with mainnet blocks, because they don't exist yet.

**Review by oxarbitrage — APPROVED** [review]
Looks great.

**Review by dconnolly — APPROVED** [review]

**Pull request merged**